src/vcpkg/commands.add.cpp

```diff
@@ -16,10 +20,18 @@ namespace
             create_example_string("add port png"),
             create_example_string("add artifact cmake")),
         2,
-        2,
+        SIZE_MAX,
         {{}, {}},
         nullptr,
     };
+
+    DECLARE_AND_REGISTER_MESSAGE(
+        AddTripletExpressionNotAllowed,
+        (),
+        "",
+        "Error: Triplet expressions are not allowed here. Use the --triplet parameter when running `vcpkg install`.");
```

Comment:
```suggestion
        "Error: Triplet expressions are not allowed here.");
```

Reply:
Why not give the user a hint how to set the triplet?

Reply:
It might be nice to give a hint as to how to do this correctly for `add`; something like

```
"Error: triplet expressions are not allowed here. You may want to change `{name}:{triplet}` to `{name}` instead."
```

Reply:
This command doesn't understand triplets at all, if I understand the situation correctly. This is the message that will be displayed upon the user writing:
```
vcpkg add port zlib:x64-windows
```

Reply:
> This command doesn't understand triplets at all, if I understand the situation correctly.

Right

Reply:
Which means that this error message is like "The oven is not allowed when making ice cream. When making cookies, be sure to set the oven to 350 degrees F." We should only be printing messages related to what the user asked to do, not unrelated things.